regxmllib/CMakeLists.txt

```diff
@@ -9,7 +9,8 @@ include_directories( ${XercesC_INCLUDE_DIR}
 
 
 file(GLOB_RECURSE SRC_FILES src/main/cpp/*.cpp src/main/cpp/*.h )
-add_library(${PROJECT_NAME} ${SRC_FILES})
+add_library(${PROJECT_NAME} SHARED ${SRC_FILES})
```

Comment:
Building of shared vs dynamic libraries can be controlled by  BUILD_SHARED_LIBS instead of hard-coding in `CMakeLists.txt`, right?

See https://stackoverflow.com/questions/2152077/is-it-possible-to-get-cmake-to-build-both-a-static-and-shared-version-of-the-sam

Reply:
This is true. I will revert back to the original code. However for the install line we would have to add the following:

if(BUILD_SHARED_LIBS)
    install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION lib)
else()
    install(TARGETS ${PROJECT_NAME} ARCHIVE DESTINATION lib)
endif()

Since static libraries are treated as "ARCHIVE" targets and shared libraries are treated as "LIBRARY" targets

See https://cmake.org/cmake/help/v3.0/command/install.html?highlight=install

Reply:
@ethanchan747 The documentation seems to imply that the if...then...else is not needed and that the following will work:

```
install(TARGETS ${PROJECT_NAME} LIBRARY DESTINATION lib ARCHIVE DESTINATION lib)
```

See https://cmake.org/cmake/help/v3.5/command/install.html

Reply:
you're right again! I just tried it out and the code supplied above works. I have submitted an updated pull request